test/common/TestBed.cpp

```diff
@@ -220,17 +220,20 @@ namespace RcclUnitTesting
     }
   }
 
-  void TestBed::ExecuteCollectives()
+  void TestBed::ExecuteCollectives(std::vector<int> const &currentRanks)
   {
     int const cmd = TestBedChild::CHILD_EXECUTE_COLL;
-    ++TestBed::NumTestsRun();
-
+    if (currentRanks.empty() || (currentRanks.size() > 1)) ++TestBed::NumTestsRun(); // this wont be accurate on the case of multiple children running send or receive
     // Send ExecuteColl command to each active child process
     for (int childId = 0; childId < this->numActiveChildren; ++childId)
     {
       PIPE_WRITE(childId, cmd);
+      int tempCurrentRanks = currentRanks.size();
+      PIPE_WRITE(childId, tempCurrentRanks);
+      for (int rank = 0; rank < currentRanks.size(); ++rank){
+        PIPE_WRITE(childId, currentRanks[rank]);
+      }
     }
```

Comment:
This doesn't look quite right.  Generally, we are passing global rank ids to children, but only to the children that own that global rank.  Each child should receive a different number of global rank ids.